internal/postgres/config.go

```diff
@@ -170,7 +170,7 @@ func Environment(cluster *v1beta1.PostgresCluster) []corev1.EnvVar {
 // reloadCommand returns an entrypoint that convinces PostgreSQL to reload
 // certificate files when they change. The process will appear as name in `ps`
 // and `top`.
-func reloadCommand(name string, post250 bool) []string {
+func reloadCommand(name string, post250 bool, AutoGrowVolumes bool) []string {
```

Comment:
I think we need to introduce a version check for changing the script in this function given that it is triggered here:

```
	reloader := corev1.Container{
		Name: naming.ContainerClientCertCopy,

		Command: reloadCommand(naming.ContainerClientCertCopy, inCluster.CompareVersion("2.5.0") >= 0, feature.Enabled(ctx, feature.AutoGrowVolumes)),

		Image:           container.Image,
		ImagePullPolicy: container.ImagePullPolicy,
		SecurityContext: initialize.RestrictedSecurityContext(inCluster.CompareVersion("2.5.0") >= 0),

		VolumeMounts: []corev1.VolumeMount{certVolumeMount},
	}
```

You can verify that by checking the failures on the `upgrade-minor` e2e test